woodwork/table_accessor.py

```diff
@@ -1615,7 +1615,12 @@ def _infer_missing_logical_types(
             null_invalid_values=null_invalid_values,
         )
         if updated_series is not series:
-            dataframe[name] = updated_series
+            # NotImplementedError thrown by dask when attempting to re-initialize
+            # data after being assigned a numeric column name
+            try:
+                dataframe[name] = updated_series
+            except NotImplementedError:
+                pass
```

Comment:
can we get around this by upgrading dask or something? Seems weird to just `pass` here

Reply:
I couldn't find a version of dask that satisfies this issue unfortunately. We can't assign a numeric column name in this specific case for dask, but I'm open to replacing it with something else though. It doesn't seem like something that we should raise an error for imo